accounting/src/delta/combine.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2021 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common::primitives::{signed_amount::SignedAmount, Amount};
+
+use crate::error::Error;
+
+use super::delta_data_collection::DataDelta;
+
+pub fn combine_data_with_delta<T: Clone>(
+    parent_data: Option<T>,
+    local_data: Option<&DataDelta<T>>,
+) -> Result<Option<T>, Error> {
+    match (parent_data, local_data) {
+        (None, None) => Ok(None),
+        (None, Some(d)) => match d {
+            DataDelta::Create(d) => Ok(Some(*d.clone())),
+            DataDelta::Modify(_) => Err(Error::ModifyNonexistingData),
+            DataDelta::Delete => Err(Error::RemoveNonexistingData),
+        },
+        (Some(p), None) => Ok(Some(p)),
+        (Some(_), Some(d)) => match d {
+            DataDelta::Create(_) => Err(Error::DataCreatedMultipleTimes),
+            DataDelta::Modify(d) => Ok(Some(*d.clone())),
+            DataDelta::Delete => Ok(None),
+        },
+    }
+}
+
+/// add two numbers that can be Some or None, one unsigned and another signed
+/// If both numbers are None, then the result is none (if key not found in both parent and local)
+/// If only unsigned is present, then the unsigned is returned (only parent found)
+/// If only signed is present, we convert it to unsigned and return it (only delta found)
+/// If both found, we add them and return them as unsigned
+/// Errors can happen when doing conversions; which can uncover inconsistency issues
+pub fn combine_amount_delta(
+    lhs: &Option<Amount>,
+    rhs: &Option<SignedAmount>,
+) -> Result<Option<Amount>, Error> {
+    match (lhs, rhs) {
+        (None, None) => Ok(None),
+        (None, Some(v)) => Ok(Some(
+            (*v).into_unsigned().ok_or(Error::ArithmeticErrorToUnsignedFailed)?,
+        )),
+        (Some(v), None) => Ok(Some(*v)),
+        (Some(v1), Some(v2)) => {
+            let v1 = v1.into_signed().ok_or(Error::ArithmeticErrorToSignedFailed)?;
+            let sum = (v1 + *v2).ok_or(Error::ArithmeticErrorDeltaAdditionFailed)?;
+            let sum = sum.into_unsigned().ok_or(Error::ArithmeticErrorSumToUnsignedFailed)?;
+            Ok(Some(sum))
+        }
+    }
+}
+
+#[cfg(test)]
+pub mod test {
+    use super::*;
+    use common::primitives::{amount::UnsignedIntType, signed_amount::SignedIntType};
+
+    #[test]
+    #[rustfmt::skip]
```

Comment:
Just curious: formatting is disabled here because rustfmt will split assertions into several lines? Is that bad? Can only that rule be overridden locally?